### PR TITLE
Extract thrift proxy stat_prefix into a prometheus tag

### PR DESCRIPTION
**Pull request by davinci26**
Commit Message:

Fixes issue #20201.

Thrift proxy stats prefix is not extracted as a tag in prometheus stats output. This makes it harder to generate generic graphs. 
Additional Description: N/A
Risk Level: Low (stats)
Testing: Unit + manual
Docs Changes: N/A
Release Notes: Pending (after we agree on the design to avoid merge conflicts)
Platform Specific Features: N/A
Optional Runtime guard:
Guard via: `envoy.restart_features.thrift_stat_prefix_tag_extraction` to maintain backwards compatibility if needed.
Signed-off-by: Sotiris Nanopoulos <sotiris.nanopoulos@reddit.com>

### CODE REVIEW

**Comment by davinci26** [comment]
Will assign the thrift reviewers when CI goes green

**Comment by davinci26** [comment]
image

output with this PR form `stats/prometheus`

cc @bjk-

**Comment by rgs1** [comment]
LGTM modulo the pending changelog entry.

cc: @jmarantz for stats related code.

File: source/common/config/well_known_names.cc
Comment (ggreenway) [inline source/common/config/well_known_names.cc:149]: Last time I tried this, (IIRC) runtime flags don't work here because of startup-ordering problems. See #17686. cc @jmarantz

**Review by ggreenway — CHANGES_REQUESTED** [review]

File: source/common/config/well_known_names.cc
Comment (davinci26) [reply source/common/config/well_known_names.cc:149]: fun stuff :) let me verify it, because I verified this with the tests

Comment (ggreenway) [reply source/common/config/well_known_names.cc:149]: It's possible that something has changed since then, and this is no longer a problem. That would be great news.

File: test/common/stats/tag_extractor_test_common.h
Comment (jmarantz) [inline test/common/stats/tag_extractor_test_common.h:26]: can you put the impls for this class into a .cc file?

File: source/common/config/well_known_names.cc
Comment (ggreenway) [reply source/common/config/well_known_names.cc:149]: Oh, I see; in my PR, I was using an integration test. I just updated that PR branch onto main and it still fails. I *think* it would also fail on a real (non-test) build due to init-order problems.

**Comment by jmarantz** [comment]
I assume this is blocked on the init-order issue described by @ggreenway and maybe needs an integration test to validate that.

Please let me know if that's not right and this is ready for review.

/wait

**Pull request closed without merging**
